src/inventory.h

```diff
@@ -89,6 +89,26 @@ class invlet_favorites
         std::array<itype_id, 256> ids_by_invlet;
 };
 
+struct quality_query {
+    quality_id qual;
+    int level;
+    int count;
+
+    bool operator==( const quality_query &other ) const {
+        return qual == other.qual && level == other.level && count == other.count;
+    }
+
+    bool operator<( const quality_query &other ) const {
+        if( qual == other.qual ) {
+            if( level == other.level ) {
+                return count < other.count;
+            }
+            return level < other.level;
+        }
+        return qual < other.qual;
+    }
```

Comment:
```suggestion
    bool operator<( const quality_query &other ) const {
        return std::tie( qual, level, count ) < std::tie( other.qual, other.level, other.count );
    }
```
It's up to you, but this could be shortened to use tuples to avoid nesting

Reply:
Hard to decide (for me as a non C++ guy). It looks really neat, but I don't think I would know what I'm looking at if I just saw that without knowing what it does. Cppreference does have an example doing exactly this, though.